Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.requestContent = function() {
+        var that = this;
+
+        if (this.hasEmptyContent) {
+            return false;
+        }
+
+        if (!this._requestServer.hasAvailableRequests()) {
+            return false;
+        }
+
+        var url = this._contentUrl;
+        if (defined(this.expireDate)) {
+            // Append a query parameter of the tile expiration date to prevent caching
+            var timestampQuery = '?expired=' + this.expireDate.toString();
+            url = joinUrls(url, timestampQuery, false);
+        }
+
+        var distance = this._distanceToCamera;
+        var promise = RequestScheduler.schedule(new Request({
+            url : url,
+            server : this._requestServer,
+            requestFunction : loadArrayBuffer,
+            type : RequestType.TILES3D,
+            distance : distance
+        }));
+
+        if (!defined(promise)) {
+            return false;
+        }
+
+        this._contentState = Cesium3DTileContentState.LOADING;
+        this._contentReadyToProcessPromise = when.defer();
+        this._contentReadyPromise = when.defer();
+
+        promise.then(function(arrayBuffer) {
+            if (that.isDestroyed()) {
+                // Tile is unloaded before the content finishes loading
+                return when.reject('tile is destroyed');
+            }
+            var uint8Array = new Uint8Array(arrayBuffer);
+            var magic = getMagic(uint8Array);
+            var contentFactory = Cesium3DTileContentFactory[magic];
+            var content;
+
+            if (defined(contentFactory)) {
+                content = contentFactory(that._tileset, that, that._contentUrl, arrayBuffer, 0);
+                that.hasRenderableContent = true;
+            } else {
+                // The content may be json instead
+                content = Cesium3DTileContentFactory.json(that._tileset, that, that._contentUrl, arrayBuffer, 0);
+                that.hasTilesetContent = true;
+            }
+
+            that._content = content;
+            that._contentState = Cesium3DTileContentState.PROCESSING;
+            that._contentReadyToProcessPromise.resolve(content);
+
+            content.readyPromise.then(function(content) {
+                if (that.isDestroyed()) {
+                    // Tile is unloaded before the content finishes processing
+                    return when.reject('tile is destroyed');
+                }
+                updateExpireDate(that);
+
+                // Refresh style for expired content
+                that.lastStyleTime = 0;
+
+                that._contentState = Cesium3DTileContentState.READY;
+                that._contentReadyPromise.resolve(content);
+            }).otherwise(function(error) {
+                that._contentState = Cesium3DTileContentState.FAILED;
+                that._contentReadyPromise.reject(error);
+            });
+        }).otherwise(function(error) {
```

Comment:
The double otherwise is pretty confusing promise usage, at least add a comment but I think we should consider rethinking the control-flow here.